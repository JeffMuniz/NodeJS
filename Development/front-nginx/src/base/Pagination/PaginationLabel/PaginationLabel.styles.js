import styled from "styled-components";
import { darkGrey, black } from "src/styles/colors";

export const Wrapper = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;
  margin: 30px auto 0;
  text-align: center;
`;

export const PageLabel = styled.div`
  font-weight: 600;
  font-size: 14px;
  line-height: 14px;
  color: ${black};
  margin-right: 12px;
  min-width: 110px;
  user-select: none;
`;

export const Action = styled.span`
  color: ${props => (props.isDisabled ? darkGrey : black)};
  cursor: ${props => (props.isDisabled ? "not-allowed" : "pointer")};
  font-size: 14px;
  font-weight: 600;
  line-height: 14px;
  margin: 0 0 0 35px;
  user-select: none;
`;
