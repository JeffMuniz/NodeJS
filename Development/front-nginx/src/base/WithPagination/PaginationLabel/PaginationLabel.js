import React from "react";
import { string, func, number, bool } from "prop-types";
import { DirectionIcon } from "@common";
import { directions } from "@enums";
import { toNumberMask } from "@utils";
import { black, disabledDarkGrey } from "src/styles/colors";
import { Wrapper, PageLabel, Action } from "./PaginationLabel.styles";

export const Pagination = ({
  id,
  totalItems,
  firstItemNumber,
  lastItemNumber,
  canGoBack,
  canGoForward,
  goForward,
  goBackward,
}) => {
  if (totalItems === 0) return null;
  return (
    <Wrapper id={id}>
      <PageLabel id={`${id}_label`}>
        {toNumberMask(firstItemNumber)}-{toNumberMask(lastItemNumber)} de{" "}
        {toNumberMask(totalItems)}
      </PageLabel>
      <Action
        id={`${id}_back`}
        isDisabled={!canGoBack}
        onClick={goBackward({
          id,
          totalItems,
          firstItemNumber,
          lastItemNumber,
          canGoBack,
          canGoForward,
          goForward,
          goBackward,
        })}
      >
        <DirectionIcon
          direction={directions.left}
          color={canGoBack ? black : disabledDarkGrey}
        />
      </Action>
      <Action
        id={`${id}_go`}
        isDisabled={!canGoForward}
        onClick={goForward({
          id,
          totalItems,
          firstItemNumber,
          lastItemNumber,
          canGoBack,
          canGoForward,
          goForward,
          goBackward,
        })}
      >
        <DirectionIcon
          direction={directions.right}
          color={canGoForward ? black : disabledDarkGrey}
        />
      </Action>
    </Wrapper>
  );
};

export default Pagination;

Pagination.propTypes = {
  id: string,
  goForward: func,
  goBackward: func,
  canGoBack: bool,
  canGoForward: bool,
  lastItemNumber: number,
  firstItemNumber: number,
  totalItems: number,
};

Pagination.defaultProps = {
  id: "pagination",
  goForward: () => null,
  goBackward: () => null,
  canGoBack: false,
  canGoForward: false,
  lastItemNumber: 0,
  firstItemNumber: 0,
  totalItems: 0,
};
