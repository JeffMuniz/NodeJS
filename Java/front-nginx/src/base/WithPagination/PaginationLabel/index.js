import PaginationLabel from "./PaginationLabel";

export default PaginationLabel;
