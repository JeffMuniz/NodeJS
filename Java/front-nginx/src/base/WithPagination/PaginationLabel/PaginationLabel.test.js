import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import PaginationLabel from "./PaginationLabel";

jest.mock("@common", () => ({
  SvgIcon: "SvgIcon",
  DirectionIcon: "DirectionIcon",
}));

describe("PaginationLabel", () => {
  it("Should render PaginationLabel without filtered data", () => {
    // given
    const renderer = new ShallowRenderer();
    const props = {
      onFoward: jest.fn(),
      onBackward: jest.fn(),
      canGoForward: jest.fn(),
      canGoBack: jest.fn(),
      lastItemNumber: 10,
      firstItemNumber: 1,
      totalItems: 100,
    };

    // when
    renderer.render(<PaginationLabel {...props} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });
});
