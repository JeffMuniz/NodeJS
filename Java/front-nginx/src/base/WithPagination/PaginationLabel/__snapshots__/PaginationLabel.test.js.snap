// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationLabel Should render PaginationLabel without filtered data 1`] = `
<styled.div
  id="pagination"
>
  <styled.div
    id="pagination_label"
  >
    1
    -
    10
     de
     
    100
  </styled.div>
  <styled.span
    id="pagination_back"
    isDisabled={false}
    onClick={null}
  >
    <DirectionIcon
      color="#000000"
      direction="left"
    />
  </styled.span>
  <styled.span
    id="pagination_go"
    isDisabled={false}
    onClick={null}
  >
    <DirectionIcon
      color="#000000"
      direction="right"
    />
  </styled.span>
</styled.div>
`;
